toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -352,6 +371,136 @@ func findPartition(mountIdType imagecustomizerapi.MountIdentifierType, mountId s
 	return partition, partitionIndex, nil
 }
 
+func findExtendedPartition(mountIdType imagecustomizerapi.ExtendedMountIdentifierType, mountId string,
+	partitions []diskutils.PartitionInfo, buildDir string,
+) (diskutils.PartitionInfo, int, error) {
+	matchedPartitionIndexes := []int(nil)
+	for i, partition := range partitions {
+		matches := false
+		switch mountIdType {
+		case imagecustomizerapi.ExtendedMountIdentifierTypeUuid:
+			matches = partition.Uuid == mountId
+		case imagecustomizerapi.ExtendedMountIdentifierTypePartUuid:
+			matches = partition.PartUuid == mountId
+		case imagecustomizerapi.ExtendedMountIdentifierTypePartLabel:
+			matches = partition.PartLabel == mountId
+		case imagecustomizerapi.ExtendedMountIdentifierTypeDev:
+			matches, err := findExtendedDevPartition(partition, partitions, buildDir)
+			if err != nil {
+				return diskutils.PartitionInfo{}, 0, err
+			}
+			if matches {
+				matchedPartitionIndexes = append(matchedPartitionIndexes, i)
+			}
+		}
+		if matches {
+			matchedPartitionIndexes = append(matchedPartitionIndexes, i)
+		}
+	}
+
+	if len(matchedPartitionIndexes) < 1 {
+		err := fmt.Errorf("partition not found (%s=%s)", mountIdType, mountId)
+		return diskutils.PartitionInfo{}, 0, err
+	}
+	if len(matchedPartitionIndexes) > 1 {
+		err := fmt.Errorf("too many matches for partition found (%s=%s)", mountIdType, mountId)
+		return diskutils.PartitionInfo{}, 0, err
+	}
+
+	partitionIndex := matchedPartitionIndexes[0]
+	partition := partitions[partitionIndex]
+
+	return partition, partitionIndex, nil
+}
+
+func findExtendedDevPartition(partition diskutils.PartitionInfo, partitions []diskutils.PartitionInfo,
+	buildDir string,
+) (bool, error) {
+	espPartition, err := findSystemBootPartition(partitions)
+	if err != nil {
+		return false, err
+	}
+
+	bootPartition, err := findBootPartitionFromEsp(espPartition, partitions, buildDir)
+	if err != nil {
+		return false, err
+	}
+
+	// Temporarily mount the boot partition so that the grub config file can be read.
+	tmpDirBoot := filepath.Join(buildDir, tmpBootPartitionDirName)
+	bootPartitionMount, err := safemount.NewMount(bootPartition.Path, tmpDirBoot, bootPartition.FileSystemType, 0, "", true)
```

Comment:
This mount should be read-only. Mounting partitions non-read-only modifies it unintentionally.